llvm/lib/IR/Verifier.cpp

```diff
@@ -842,7 +842,7 @@ void Verifier::visitGlobalVariable(const GlobalVariable &GV) {
     // If the global has common linkage, it must have a zero initializer and
     // cannot be constant.
     if (GV.hasCommonLinkage()) {
-      Check(GV.getInitializer()->isNullValue(),
+      Check(GV.getInitializer()->isZeroValue(),
```

Comment:
I wonder if we should add an optional data layout parameter to this function so it can return true for null pointers in the common case?